library/spdm_secured_message_lib/libspdm_secmes_encode_decode.c

```diff
@@ -129,7 +117,9 @@ libspdm_return_t libspdm_encode_secured_message(
         return LIBSPDM_STATUS_SEQUENCE_NUMBER_OVERFLOW;
     }
 
-    *(uint64_t *)salt = *(uint64_t *)salt ^ sequence_number;
+    if (sequence_number > 0) {
+        *(uint64_t *)salt = *(uint64_t *)salt ^ (sequence_number - 1) ^ sequence_number;
```

Comment:
This (and the 2.3 version) is misleading as once the assignment occurs this value is no longer the `salt`, it's the `IV`. So here I would zero-extend `sequence_number` and XOR it with `salt` and assign it (libspdm_copy_mem) to `iv[]`.

Reply:
If the name is confusing, I think we can rename to `uint8_t *iv`. I don't see the need to cache the `iv[]`.